[bug] Current vcpkg compilation is not working out of the box

I recently installed vcpkg on windows 10 along with VS2019 community.

I ran:

`vcpkg install cctag[cuda,apps]:x64-windows`

It crashes even after manually applying the fix for the known issue #166 

The problem appears : 
`Creating library src\CCTagd.lib and object src\CCTagd.exp
Bresenham.cpp.obj : error LNK2019: unresolved external symbol "int __cdecl boost::math::sign<int>(int const &)" (??$sign@H@math@boost@@YAHAEBH@Z) referenced in function "void __cdecl cctag::updateXY(float const &,float const &,int &,int &,float &,int &,int &)" (?updateXY@cctag@@YAXAEBM0AEAH1AEAM11@Z)
Windows-AMD64\CCTagd.dll : fatal error LNK1120: 1 unresolved externals
ninja: build stopped: subcommand failed.`

Is seems to be linked with issue #166 that added the line in Bresenham.cpp:
`#include <boost/math/special_functions/sign.hpp>`
But without informing CMAKE that boost::math is used.


**Desktop:**
 - OS: Windows 10 x64
 - Cuda version CUDA V11.4.100
 - Boost 1.76
 - CCTag version: 1.0.0 from the vcpkg repository

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -110,7 +110,7 @@ if(APPLE)
 endif(APPLE)
 
 # FIND BOOST
-set(BOOST_REQUIRED_COMPONENTS "atomic;chrono;date_time;filesystem;program_options;serialization;system;thread;timer")
+set(BOOST_REQUIRED_COMPONENTS "atomic;chrono;date_time;filesystem;program_options;serialization;system;thread;timer;math_c99")
 if(WIN32)
   set(BOOST_REQUIRED_COMPONENTS "${BOOST_REQUIRED_COMPONENTS};stacktrace_windbg")
 else()

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -227,7 +227,7 @@ else() # without CUDA
                         PUBLIC
                           ${OpenCV_LIBS}
                           Eigen3::Eigen
-                          Boost::atomic Boost::chrono Boost::date_time Boost::filesystem Boost::serialization Boost::system Boost::thread Boost::timer
+                          Boost::atomic Boost::chrono Boost::date_time Boost::filesystem Boost::serialization Boost::system Boost::thread Boost::timer Boost::math_c99
                         PRIVATE
                           ${TBB_LIBRARIES})
 

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -174,7 +174,7 @@ if(CCTAG_WITH_CUDA)
   # since we used CUDA_ADD_LIBRARY we cannot use PUBLIC or PRIVATE here
   target_link_libraries(CCTag
                        ${OpenCV_LIBS}
-                       Boost::date_time Boost::chrono Boost::thread Boost::serialization Boost::system Boost::filesystem Boost::atomic Boost::program_options Boost::timer
+                       Boost::date_time Boost::chrono Boost::thread Boost::serialization Boost::system Boost::filesystem Boost::atomic Boost::program_options Boost::timer Boost::math_c99
                        Eigen3::Eigen
                        ${TBB_LIBRARIES} ${CUDA_CUDADEVRT_LIBRARY})
 

